connectivity/service/src/main/java/org/eclipse/ditto/connectivity/service/messaging/internal/ssl/Keys.java

```diff
@@ -115,10 +115,11 @@ private static <T extends KeySpec> T matchKey(final Matcher matcher, final Funct
      * @param exceptionMapper maps ssl exceptions to ditto exceptions providing context information
      * @return the certificate
      */
-    static Certificate getCertificate(final String certificatePem, final ExceptionMapper exceptionMapper) {
+    static Certificate[] getCertificateChain(final String certificatePem, final ExceptionMapper exceptionMapper) {
         final byte[] asciiBytes = certificatePem.getBytes(StandardCharsets.US_ASCII);
         try {
-            return X509_CERTIFICATE_FACTORY.generateCertificate(new ByteArrayInputStream(asciiBytes));
+            return X509_CERTIFICATE_FACTORY.generateCertificates(new ByteArrayInputStream(asciiBytes))
```

Comment:
This `X509_CERTIFICATE_FACTORY.generateCertificates` does the "parsing of the full certificate chain", right?

Because returned and used is also only the client certificate, right?

Reply:
@JCapucho ok .. but you only select the first element of the returned certificates chain to be used:
```
keyStoreFactory.setCertificate(keystore, certificate[0]);
```

That's what I meant with that still only the client certificate - and not the chain - is used.

Reply:
I see what you mean now, as far as i understand `setCertificateEntry` sets a trusted certificate for the keystore, which in the context of client certificate authentication is not relevant. What's actually relevant is the `setPrivateKey` line above it, which sets the private key and the **certificate chain** used for authentication, and as you can see uses the full certificate chain returned from the method.

I'm going to be honest I don't quite understand the purpose of the `setCertificate` line so I just left it with it's original behavior.